tests/tensorflow/r2.12/common.libsonnet

```diff
@@ -23,9 +23,62 @@ local mixins = import 'templates/mixins.libsonnet';
 
     frameworkPrefix: 'tf-r2.12.0',
     tpuSettings+: {
-      softwareVersion: '2.12.0',
+      softwareVersion: '2.11.0',
     },
     imageTag: 'r2.12.0',
+    podTemplate+:: {
+      spec+: {
+        initContainerMap+:: {
+          'tpu-version': {
+            image: config.podTemplate.spec.containerMap.train.image,
+            env+: [
+              {
+                name: 'TPU_NAME',
+                valueFrom: {
+                  fieldRef: {
+                    fieldPath: "metadata.annotations['name.cloud-tpus.google.com/train']",
+                  },
+                },
+              },
+              {
+                name: 'POD_UID',
+                valueFrom: {
+                  fieldRef: {
+                    fieldPath: 'metadata.uid',
+                  },
+                },
+              },
+            ],
+            local tpuCreateSettings = {
+              acceleratorName: std.escapeStringBash(config.accelerator.name),
+              softwareVersion: std.escapeStringBash(config.tpuSettings.softwareVersion),
+              startupScript: std.escapeStringBash(config.tpuSettings.tpuVmStartupScript),
+              sleepTime: config.tpuSettings.tpuVmCreateSleepSeconds,
+              testName: std.strReplace(config.testName, '.', '-'),
+            },
+            command: [
```

Comment:
Would this prepend an existing command, or overwrite it altogether? Or does that even matter in this context?

Reply:
I belive this would overwrite an existing command but to my knowledge there is not one that exists, so it does not matter. The change in this PR is similar to what pytorch uses for 2vm: https://github.com/GoogleCloudPlatform/ml-testing-accelerators/blob/master/tests/pytorch/nightly/common.libsonnet#L47-L73